### DISCUSSION TITLE
Graphiql integration

### DISCUSSION THREAD

**Discussion by kefniark** [enisdenjo/graphql-sse#8] [Q&A] [answered]
## Description
So far I'm quite happy with `graphql-sse`, it's doing the job both frontend and backend side.

But there is one thing that annoy me a bit for the dev experience, the fact that is doesn't look compatible with `graphiql` by default. For subscriptions, they support `graphql-ws` or apollo web socket, they support `sse` subscriptions but sadly it doesn't work with `graphql-sse`

The way they do SSE queries seem to be not recognized by the server (as GET with query parameter), and I get 404 error `Stream not found`
![image](https://user-images.githubusercontent.com/4903529/143383091-34e73f97-caa7-44e5-98e9-1d4481e61f22.png)

I found [some gist](https://gist.github.com/enisdenjo/d7bc1a013433502349d2763c3d2f2b79), how to modify graphiql html page to add `graphql-sse` client, but it's not always available (for example `graphql-helix` only expose a limited set of configurations).

Would it be possible server side, at the createHandler level, to parse this query format and modify the request to make it compatible with what `graphql-sse` expect ?

**Comment by enisdenjo** [comment, marked as answer]
Sorry for the late reply. 😅

`graphql-helix` does not implement `graphql-sse`'s [GraphQL over SSE Protocol](https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md) - is therefore not compatible.

The [Gist you referenced](https://gist.github.com/enisdenjo/d7bc1a013433502349d2763c3d2f2b79) should work with `graphql-sse` powered servers without any problems! I don't quite understand what do you mean by: _"Would it be possible server side, at the createHandler level, to parse this query format and modify the request to make it compatible with what graphql-sse expect ?"_. Can please make me a repro so that I can understand what's happening?
